### PR TITLE
Add support for matrix builds over GC implementation

### PR DESCRIPTION
**Pull request by densh**

### CODE REVIEW

File: bin/travis_mac_setup.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+rew update
```
Comment (jonas) [inline bin/travis_mac_setup.sh]: Also adding a proper `#!` header would make it usable as a stand-alone script for installing tools.

File: .travis.yml [hunk 2]
```diff
@@ -8,47 +8,34 @@ matrix:
...
+        - SCALANATIVE_GC=boehm
+        - CLANG_FORMAT_PATH="/usr/local/opt/llvm@4/bin/clang-format"
+      before_install:
+        - bash bin/travis_mac_setup.sh
 
     - os: osx
       osx_image: xcode7.3
+      env:
+        - SCALANATIVE_GC=none
+        - CLANG_FORMAT_PATH="/usr/local/opt/llvm@4/bin/clang-format"
```
Comment (jonas) [inline .travis.yml]: `travis lint` complains that you can only define one env var per matrix entry. One option would be to to use something else than an env variable to trigger the formatting check or start using build stages https://docs.travis-ci.com/user/build-stages/warm-cache/

File: bin/travis_mac_setup.sh (hunk 1 above)
Comment (densh) [inline bin/travis_mac_setup.sh]: Nice catch and good idea!

File: .travis.yml (hunk 2 above)
Comment (densh) [inline .travis.yml]: Nice catch! We can set CLANG_FORMAT_PATH for every platform but run it only if it exists.

**Pull request merged**